kong/db/schema/entities/routes_subschemas.lua

```diff
@@ -67,17 +67,39 @@ local grpc_subschema = {
 
 
 if kong and kong.configuration and  kong.configuration.router_flavor == "expressions" then
-  return {}
-
-else
-  return {
-    http = http_subschema,  -- protocols is the subschema key, and the first
-    https = http_subschema, -- matching protocol name is selected as subschema name
-    tcp = stream_subschema,
-    tls = stream_subschema,
-    udp = stream_subschema,
-    tls_passthrough = stream_subschema,
-    grpc = grpc_subschema,
-    grpcs = grpc_subschema,
-  }
+
+  -- now http route in flavor 'expressions' accepts 'sources' 'and destinations'
+
+  http_subschema.fields[1] = nil  -- sources
+  http_subschema.fields[2] = nil  -- destinations
+
+  -- the route should have the field 'expression' if no others
+
+  table.insert(http_subschema.entity_checks[1].conditional_at_least_one_of.then_at_least_one_of, "expression")
+  table.insert(http_subschema.entity_checks[1].conditional_at_least_one_of.else_then_at_least_one_of, "expression")
+
+  -- now grpc route in flavor 'expressions' accepts 'sources' 'and destinations'
+
+  grpc_subschema.fields[3] = nil  -- sources
+  grpc_subschema.fields[4] = nil  -- destinations
```

Comment:
We should add assertions here to ensure we are deleting expected fields.  We might insert some new fields in the future and forget to correct the index.